src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -931,42 +931,74 @@ export function mergeValidationObjects(...sources: IValidations[]): IValidations
 
   sources.forEach((source: IValidations) => {
     Object.keys(source).forEach((layout: string) => {
-      if (!validations[layout]) {
-        validations[layout] = {};
-      }
-      Object.keys(source[layout] || {}).forEach((component: string) => {
-        if (!validations[layout][component]) {
-          validations[layout][component] = {};
-        }
-        Object.keys(source[layout][component] || {}).forEach((binding: string) => {
-          if (!validations[layout][component][binding]) {
-            validations[layout][component][binding] = { errors: [], warnings: [] };
-          }
-
-          // Only merge items that are not already in the existing components errors/warnings array
-          const newErrors = source[layout][component][binding]?.errors?.filter((error) => {
-            return validations[layout][component][binding]?.errors.findIndex((existingError) => {
-              return JSON.stringify(existingError) === JSON.stringify(error);
-            }) < 0;
-          });
-          validations[layout][component][binding].errors =
-            validations[layout][component][binding]?.errors.concat(newErrors);
-
-          const newWarnings = source[layout][component][binding]?.warnings?.filter((warning) => {
-            return validations[layout][component][binding]?.warnings.findIndex((existingWarning) => {
-              return JSON.stringify(existingWarning) === JSON.stringify(warning);
-            }) < 0;
-          });
-          validations[layout][component][binding].warnings =
-            validations[layout][component][binding]?.warnings.concat(newWarnings);
-        });
-      });
+      validations[layout] = mergeLayoutValidations(validations[layout] || {}, source[layout] || {});
     });
   });
 
   return validations;
 }
 
+export function mergeLayoutValidations(
```

Comment:
Nice!
Are the functions that are used by mergeLayoutValidation exported because of testing?
We could possibly try to nest these into a module directory at some point and only export what's actually used by validations.ts.